### PR TITLE
Add LocalStack (V1) container for zio 2.0

### PR DESCRIPTION
**Pull request by grouzen**
This PR adds LocalStack container V1 for ZIO 2.0 only. 

~I also opened another PR in `testcontainers-scala` to allow `dockerImageName` overrides, and not only `tag`: https://github.com/testcontainers/testcontainers-scala/pull/246. I think it would be good to wait until the `testcontainers-scala` PR is merged and make changes to this one accordingly.~
Update: `testcontainers-scala` is bumped up to `0.40.15`

### CODE REVIEW

**Comment by grouzen** [comment]
I'm going to add documentation if this approach is fine.

File: modules/localstack-zio-2.0/src/test/scala/io/github/scottweaver/zio/testcontainers/localstack/ZLocalStackContainerSpec.scala [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+      AwsConfig.configured(),
+      commonAwsConfigLayer,
+      NettyHttpClient.default
+    )
+
+  private def s3Layer: RLayer[LocalStackContainer with AwsConfig, S3] =
+    ZLayer.scoped {
+      for {
+        localstack <- ZIO.service[LocalStackContainer]
+        endpoint   <- ZIO.attempt(localstack.container.getEndpointOverride(JavaLocalStackContainer.Service.S3))
```
Comment (grouzen) [inline modules/localstack-zio-2.0/src/test/scala/io/github/scottweaver/zio/testcontainers/localstack/ZLocalStackContainerSpec.scala:59]: This shows an example of overriding endpoints of LocalStack services.

File: build.sbt [hunk 2]
```diff
@@ -269,6 +270,29 @@ lazy val solrZio2 =
...
+  project
+    .in(file("modules/localstack-zio-2.0"))
+    .settings(
+      zioSeries := ZIOSeries.Series2X,
+      testcontainersScalaSettings,
+      name := "zio-2.0-testcontainers-localstack",
+      libraryDependencies ++= Seq(
+        "com.dimafeng" %% "testcontainers-scala-localstack" % V.testcontainersScalaVersion,
+        "com.amazonaws" % "aws-java-sdk-s3"                 % V.awsV1Version  % Provided,
+        "com.amazonaws" % "aws-java-sdk-sqs"                % V.awsV1Version  % Provided,
```
Comment (grouzen) [inline build.sbt:283]: That's the way `testcontainers-scala` provides these dependencies. I decided to stick to their approach.

File: project/V.scala [hunk 3]
```diff
@@ -12,6 +12,8 @@ object V {
   val zioKafkaVersion            = "0.17.1"
   val zio2KafkaVersion           = "2.0.0"
   val zioHttpVersion             = "0.0.4"
+  val zioAwsVersion              = "5.20.17.1"
```
Comment (grouzen) [inline project/V.scala:15]: I believe this is the latest version of `zio-aws` that uses ZIO `2.0.9`. Usage of more recent versions causes `NoClassDefFoundError`. It should be updated after the ZIO version is bumped up.

**Comment by grouzen** [comment]
Fixing the compilation error...

**Comment by V-Lamp** [comment]
Hi, I have implemented something similar and opted to directly integrate it with zio-aws. 
It does work and allows for parallel testing, which plays great with zio-test by default parallelism.

The interesting/useful thing is on how to get the actual AWS config as it turns out that you don't need to have service-specific endpoints, therefore you don't need to pass the Service enum value in the TestContainer API to get an endpoint and instead get a layer to be reused by all services, which is great if you have code using multiple services. You just provide the layer 😃 

Even though may not want to depend on zio-aws, it would be great to expose the endpoint directly to simplify the usage, as most users will think they need to pass the Service enum to get the endpoint.

```scala
  val awsConfig: ZLayer[Any, Throwable, CommonAwsConfig] = ZLayer.scoped {
    for {
      localstack <- makeContainer(LocalStackV2Container.Def(
        tag = "2.0.2",
        // Services are loaded lazily in latest localstack so need to be specific.
        services = Service.values()
      ))
      hostIpAddress <- ZIO.attempt(InetAddress.getByName(localstack.host).getHostAddress)
      // All services are found in the same port in latest locastack, so no service-specific mapping needed
      port <- ZIO.attempt(localstack.mappedPort(4566))
      uri <- ZIO.attempt(new URI(s"http://${hostIpAddress}:${port}"))
    } yield CommonAwsConfig(
      region = Some(Region.EU_WEST_1),
      endpointOverride = Some(uri),
      credentialsProvider = localstack.staticCredentialsProvider,
      commonClientConfig = None
    )
  }
  ```
Hope it helps !

**TLDR**: Updates to Localstack make the API container API from testcontainers outdated, as you don't need to either specify which services you need, nor get service-specific ports. So the `org.testcontainers.containers.localstack.LocalStackContainer.Service` enum becomes pretty much useless.

Dropping the `withServices` and `endpointOverride` methods (as there are very limited reasons to not use the latest localstack version) will help users avoid a legacy API which leads to quite complicated usage, especially when using zio layers.
